Add support for color

FokkeZB opened the pull request:
Closes #34

Commit:
link color fix

Commit:
link color fix

Commit:
link color

FokkeZB commented on index.js:
```diff
@@ -105,8 +105,16 @@ function walker(node, parameters, outerFont, customMatcher) {
+            value: node.attribs.href,
+            range: [offset, length]
+        }, {
+          type : ns.ATTRIBUTE_FOREGROUND_COLOR,
```
Lines 112-117 should be wrapped in a if checking if `node.attrib.color` is set.

samueleastdev replied:
Hi Fokke,
I have added an if statement and extra info to the readme that would help users when adding html to the string.xml doc using cdata. I can't seem to get the indexSpec.js to run so I have just taken out what I added I get the error below when running grunt.

~~~
[DEBUG] module.js:327
[DEBUG]     throw err;
[DEBUG]     ^
[ERROR] : Cannot find module '/Users/sam.east/Desktop/ti-html2as/undefined'
[DEBUG]     at Function.Module._resolveFilename (module.js:325:15)
[DEBUG]     at Function.Module._load (module.js:276:25)
[DEBUG]     at Function.Module.runMain (module.js:441:10)
[DEBUG]     at startup (node.js:139:18)
[DEBUG]     at node.js:974:3
[ERROR] Alloy compiler failed
~~~

FokkeZB reviewed (commented):
The test fails (https://travis-ci.org/FokkeZB/ti-html2as/builds/172892874) because the change is incomplete. The verification of the parsed HTML should also be checked if the attributes have been added.

FokkeZB commented:
I've created the PR for you @samueleastdev, but the code needs a few more changes. The test fails and color should only be added if the attribute exists.

Commit:
if statement check

Commit:
if statement check

Commit:
if statement check

Commit:
if statement check

Commit:
if statement check

FokkeZB commented on index.js:
```diff
@@ -104,11 +104,31 @@ function walker(node, parameters, outerFont, customMatcher) {
-          range: [offset, length]
-        });
+        
+        var linkAtributes = [{
```
Don't create a sub array. Just leave this as-is here...

FokkeZB commented on index.js:
```diff
@@ -104,11 +104,31 @@ function walker(node, parameters, outerFont, customMatcher) {
+        // check if node has a color set
+        if(node.attribs.color){
+
+          linkAtributes.push({
```
Use `parameters.attributes.unshift`

FokkeZB commented on index.js:
```diff
@@ -104,11 +104,31 @@ function walker(node, parameters, outerFont, customMatcher) {
+            range: [offset, length]
+          });
+
+          linkAtributes.push({
```
Same

FokkeZB commented on index.js:
```diff
@@ -104,11 +104,31 @@ function walker(node, parameters, outerFont, customMatcher) {
+
+        }
+
+        parameters.attributes.unshift(linkAtributes);
```
Remove, push directly to it.

FokkeZB reviewed (commented):
Changes needed

FokkeZB commented:
@samueleastdev, to run the test you have to install the dependencies once (`npm install`) and then run `npm test`. This will show you which tests failed and why, e.g.:

```
Failures:
1) html2as - standard functioinality - correctly parses known html tags to the correct AttributedString values - iOS - Ti 5.2.0
  Message:
    Expected Object({ text: 'Strong Text
    Bold Text
    Underlined Text
    Emphasised Text
    Italicised Text
    Struck (strike) Text
    Deleted Text
    Struck (s) Text
    Arial Text
    Font size 12 Text
    Red Text
    Anchor
    Letterpress Text
    Kerned Text
    Expanded Text', attributes: [ Object({ type: 20, value: '0.5', range: [ 194, 13 ] }), Object({ type: 5, value: '10', range: [ 182, 11 ] }), Object({ type: 13, value: '_UIKitNewLetterpressStyle', range: [ 165, 16 ] }), [ Object({ type: 15, value: 'https://github.com/FokkeZB/ti-html2as', range: [ 158, 6 ] }) ], Object({ type: 2, value: 'red', range: [ 149, 8 ] }), Object({ type: 0, value: Object({ fontSize: '12' }), range: [ 131, 17 ] }), Object({ type: 0, value: Object({ fontFamily: 'Arial' }), range: [ 120, 10 ] }), Object({ type: 6, value: 1, range: [ 104, 15 ] }), Object({ type: 6, value: 1, range: [ 91, 12 ] }), Object({ type: 6, value: 1, range: [ 70, 20 ] }), Object({ type: 19, value: 0.25, range: [ 54, 15 ] }), Object({ type: 19, value: 0.25, range: [ 38, 15 ] }), Object({ type: 7, value: 1, range: [ 22, 15 ] }), Object({ type: 0, value: Object({ fontWeight: 'bold' }), range: [ 12, 9 ] }), Object({ type: 0, value: Object({ fontWeight: 'bold' }), range: [ 0, 11 ] }) ] }) to equal Object({ text: 'Strong Text
    Bold Text
    Underlined Text
    Emphasised Text
    Italicised Text
    Struck (strike) Text
    Deleted Text
    Struck (s) Text
    Arial Text
    Font size 12 Text
    Red Text
    Anchor
    Letterpress Text
    Kerned Text
    Expanded Text', attributes: [ Object({ type: 20, value: '0.5', range: [ 194, 13 ] }), Object({ type: 5, value: '10', range: [ 182, 11 ] }), Object({ type: 13, value: '_UIKitNewLetterpressStyle', range: [ 165, 16 ] }), Object({ type: 15, value: 'https://github.com/FokkeZB/ti-html2as', range: [ 158, 6 ] }), Object({ type: 2, value: 'red', range: [ 149, 8 ] }), Object({ type: 0, value: Object({ fontSize: '12' }), range: [ 131, 17 ] }), Object({ type: 0, value: Object({ fontFamily: 'Arial' }), range: [ 120, 10 ] }), Object({ type: 6, value: 1, range: [ 104, 15 ] }), Object({ type: 6, value: 1, range: [ 91, 12 ] }), Object({ type: 6, value: 1, range: [ 70, 20 ] }), Object({ type: 19, value: 0.25, range: [ 54, 15 ] }), Object({ type: 19, value: 0.25, range: [ 38, 15 ] }), Object({ type: 7, value: 1, range: [ 22, 15 ] }), Object({ type: 0, value: Object({ fontWeight: 'bold' }), range: [ 12, 9 ] }), Object({ type: 0, value: Object({ fontWeight: 'bold' }), range: [ 0, 11 ] }) ] }).
```

Which fails because your changes wrap `Object({ type: 15, value: 'https://github.com/FokkeZB/ti-html2as', range: [ 158, 6 ] })` in an array, which it should not.

FokkeZB commented:
Closing for inactivity.
